Graph Services sequence diagram

jachro opened the pull request:
closes https://github.com/SwissDataScienceCenter/renku-graph/issues/39

rokroskar commented on docs/_static/uml/graph-push-event-sequence.uml:
```diff
@@ -0,0 +1,41 @@
+    deactivate Tokens
+    group Not processed commits finding process
+    Webhook->Webhook ++ : find commits ids not in the Event Log
+    Webhook->Log: find the youngest event
```
```suggestion
    Webhook->Log: find the most recent event
```

rokroskar commented on docs/_static/uml/graph-push-event-sequence.uml:
```diff
@@ -0,0 +1,41 @@
+    Webhook->Webhook ++ : find commits ids not in the Event Log
+    Webhook->Log: find the youngest event
+    activate Log
+    Log->Webhook: EventId of the youngest event or nothing
```
```suggestion
    Log->Webhook: EventId of the most recent event or nothing
```

rokroskar commented on docs/_static/uml/graph-push-event-sequence.uml:
```diff
@@ -0,0 +1,41 @@
+    activate GitLab
+    GitLab->Webhook: OK with commit info
+    deactivate GitLab
+    Webhook->Log: filter out parent commits which are already in the Event Log yet
```
```suggestion
    Webhook->Log: filter out parent commits which are already in the Event Log
```

rokroskar commented on docs/_static/uml/graph-push-event-sequence.uml:
```diff
@@ -0,0 +1,41 @@
+    activate Log
+    Log->Webhook: Parent commitIds not in the Event Log
+    deactivate Log
+    return Commit Events not in the Event Log yet
```
```suggestion
    return Commit Events not found in the Event Log
```

rokroskar commented on docs/developer/services/graph_service.rst:
```diff
@@ -17,3 +17,30 @@ architecture is illustrated below.
+
+POST <webhook-service>/projects/:id/webhooks
+
+.. uml:: ../../_static/uml/graph-create-hook-sequence.uml
```
I think putting "create webhook" before "validate webhook" would make sense

rokroskar commented on docs/developer/services/graph_service.rst:
```diff
@@ -17,3 +17,30 @@ architecture is illustrated below.
+
+.. uml:: ../../_static/uml/graph-events-status-sequence.uml
+
+POST <webhook-service>/webhooks/events
```
the "events" endpoint should come before "status"

rokroskar commented on docs/_static/uml/graph-create-hook-sequence.uml:
```diff
@@ -0,0 +1,44 @@
+    == Webhook creation - hook does exists ==
+    Client->Webhook: POST /projects/:id/webhooks
+    activate Webhook
+    ref over Webhook, Log: Hook Validation as presented above returning hook exists
```
```suggestion
    ref over Webhook, Log: Hook Validation as presented above returning OK
```

rokroskar commented on docs/_static/uml/graph-create-hook-sequence.uml:
```diff
@@ -0,0 +1,44 @@
+    == Webhook creation - hook does not exist ==
+    Client->Webhook: POST /projects/:id/webhooks
+    activate Webhook
+    ref over Webhook, Log: Hook Validation as presented above returning hook does not exist
```
```suggestion
    ref over Webhook, Log: Hook Validation as presented above returning NOT_FOUND
```

rokroskar commented on docs/_static/uml/graph-create-hook-sequence.uml:
```diff
@@ -0,0 +1,44 @@
+    activate Tokens
+    Tokens->Webhook: NO_CONTENT
+    deactivate Tokens
+    group Events history loading
```
does this mean that the hook creation triggers event processing?

jachro commented:
I addressed all the comments and in terms of the status codes I somehow find it more appropriate to use wording instead of "magic numbers" :) What I've just done is I put codes next to the names in e.g. OK (200).
